core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageCatalogService.java

```diff
@@ -279,14 +279,19 @@ private void setImageCatalogAsDefault(ImageCatalog imageCatalog) {
 
     public ImageCatalog update(ImageCatalog source) {
 
-        ImageCatalog imageCatalog = imageCatalogRepository.findOne(source.getId());
+        ImageCatalog imageCatalog = findImageCatalog(source.getId());
         authorizationService.hasReadPermission(imageCatalog);
         checkImageCatalog(imageCatalog, source.getId());
         imageCatalog.setImageCatalogName(source.getImageCatalogName());
         imageCatalog.setImageCatalogUrl(source.getImageCatalogUrl());
         return create(imageCatalog);
     }
 
+    private ImageCatalog findImageCatalog(Long id) {
+        Optional<ImageCatalog> catalogOptional = imageCatalogRepository.findById(id);
```

Comment:
This style is too long. Would be better something like this: `return imageCatalogRepository.findById(id).orElseThrow(notFound("Image catalog", id));`